src/streamer.rs

```diff
@@ -263,8 +337,25 @@ fn broadcast(
     #[cfg(feature = "erasure")]
     erasure::generate_codes(blobs);
     Crdt::broadcast(crdt, &blobs, &sock, transmit_index)?;
-    while let Some(b) = blobs.pop() {
-        recycler.recycle(b);
+    // keep the cache of blobs that are broadcast
+    {
+        let mut win = window.write().unwrap();
+        for b in &blobs {
+            let ix = b.read().unwrap().get_index().expect("blob index");
+            let pos = (ix as usize) % WINDOW_SIZE;
+            if let Some(x) = &win[pos] {
+                warn!("popped {} at {}", x.read().unwrap().get_index().unwrap(), pos);
+                recycler.recycle(x.clone());
+            }
+            win[pos] = None;
+        }
+        while let Some(b) = blobs.pop() {
+            let ix = b.read().unwrap().get_index().expect("blob index");
+            let pos = (ix as usize) % WINDOW_SIZE;
+            assert!(win[pos].is_none());
```

Comment:
@garious can you help us figure out how this could fail?  on line 342 this structure is locked, so all the blobs should be set to None